nifi-nar-bundles/nifi-hadoop-bundle/nifi-hdfs-processors/src/main/java/org/apache/nifi/processors/hadoop/GetHDFSFileInfo.java

```diff
@@ -0,0 +1,803 @@
+        props.add(FULL_PATH);
+        props.add(RECURSE_SUBDIRS);
+        props.add(DIR_FILTER);
+        props.add(FILE_FILTER);
+        props.add(FILE_EXCLUDE_FILTER);
+        props.add(IGNORE_DOTTED_DIRS);
+        props.add(IGNORE_DOTTED_FILES);
+        props.add(GROUPING);
+        props.add(DESTINATION);
+        return props;
+    }
+
+    @Override
+    public Set<Relationship> getRelationships() {
+        final Set<Relationship> relationships = new HashSet<>();
+        relationships.add(REL_SUCCESS);
+        relationships.add(REL_NOT_FOUND);
+        relationships.add(REL_ORIGINAL);
+        relationships.add(REL_FAILURE);
+        return relationships;
+    }
+
+    @Override
+    protected Collection<ValidationResult> customValidate(ValidationContext context) {
+        return super.customValidate(context);
+    }
+
+    @Override
+    public void onPropertyModified(PropertyDescriptor descriptor, String oldValue, String newValue) {
+        super.onPropertyModified(descriptor, oldValue, newValue);
+        // drop request details to rebuild it
+        req = null;
+    }
+
+    @Override
+    public void onTrigger(final ProcessContext context, final ProcessSession session) throws ProcessException {
+        FlowFile ff = null;
+        if (context.hasIncomingConnection()) {
+            ff = session.get();
+
+            // If we have no FlowFile, and all incoming connections are self-loops then we can continue on.
+            // However, if we have no FlowFile and we have connections coming from other Processors, then
+            // we know that we should run only if we have a FlowFile.
+            if (ff == null && context.hasNonLoopConnection()) {
+                context.yield();
+                return;
+            }
+        }
+        boolean scheduledFF = false;
+        if (ff == null) {
+            ff = session.create();
+            scheduledFF = true;
+        }
+        try {
+            if (req == null) {
+                //rebuild the request details based on
+                req = buildRequestDetails(context, session, ff);
+            }else {
+                //avoid rebuilding req object's patterns in order to have better performance
+                req = updateRequestDetails(context, session, ff);
+            }
+        }catch(Exception e) {
+            getLogger().error("Invalid properties: ", e);
+            if (scheduledFF) {
+                session.remove(ff);
+                context.yield();
+            }else {
+                ff = session.penalize(ff);
+                session.rollback();
+            }
+            return;
+        }
+        try {
+            final FileSystem hdfs = getFileSystem();
+            UserGroupInformation ugi = getUserGroupInformation();
+            HDFSObjectInfoDetails res = walkHDFSTree(context, session, ff, hdfs, ugi, req, null, false);
+            if (res == null) {
+                session.transfer(ff, REL_NOT_FOUND);
```

Comment:
@bbende , good catch. although, in such case there will be properties of the processor used to compute path, and other params, it would still be nice to have computed values at run time. adding this info now.